datafusion/core/tests/sqllogictests/test_files/set_variable.slt

```diff
@@ -193,31 +193,31 @@ SET TIME ZONE = '+08:00'
 query P
 SELECT '2000-01-01T00:00:00'::TIMESTAMP::TIMESTAMPTZ
 ----
-2000-01-01T08:00:00+08:00
+1999-12-31T16:00:00+08:00
```

Comment:
This is the result of https://github.com/apache/arrow-rs/pull/4201

Although I'm not entirely sure it is correct... :thinking:

Reply:
Ok I am fairly certain this is a bug in the scalar value logic

```
❯ create table foo (ts text) as values ('2000-01-01T00:00:00');

❯ select * from foo;
+---------------------+
| ts                  |
+---------------------+
| 2000-01-01T00:00:00 |
+---------------------+

# This is correct
❯ select arrow_cast(ts, 'Timestamp(Nanosecond, Some( "+08:00" ))') from foo;
+---------------------------+
| foo.ts                    |
+---------------------------+
| 2000-01-01T00:00:00+08:00 |
+---------------------------+

# This is incorrect
❯ select arrow_cast('2000-01-01T00:00:00', 'Timestamp(Nanosecond, Some( "+08:00" ))') from foo;
+-----------------------------+
| Utf8("2000-01-01T00:00:00") |
+-----------------------------+
| 1999-12-31T16:00:00+08:00   |
+-----------------------------+
1 row in set. Query took 0.002 seconds.
```

Reply:
Yes, I think a bunch of that logic is replicated in scalar.rs (as it wasn't available upstream in arrow-rs)

I think @berkaysynnada  may have some thoughts / be thinking about this.